### ISSUE TITLE
Fix `data_source="both"` in metric displays for cv reports

### ISSUE THREAD

**Issue by GaetandeCast** [probabl-ai/skore#2925]
Follow up to #2549 which was only for the `MetricsSummaryDisplay`.

These are the displays to update: 
- [x] RocCurveDisplay #3006
- [x] PrecisionRecallCurveDisplay #3006
- [ ] PredictionErrorDisplay

**Comment by GaetandeCast** [comment]
Okay so actually, since the removal of `data_source="X_y"`, displays accept `data_source="both"` but there are some issue because the plotting code assumes that `data_source="both"` is only for estimator reports.

**Comment by direkkakkar319-ops** [comment]
Hi @GaetandeCast  do we need to do separate pr for each or just one pr ??

**Comment by GaetandeCast** [comment]
maybe start with roc curve and precision recall since the fix should be similar. If prediction error is simple, you can add it too. FYI, here are the issues I found with data_source="both" for cv reports:

- **Inverted dash styles:** With `data_source="both"`, the legend shows train=dashed / test=solid, but the plotted curves often have it the other way around (especially on `CrossValidationReport`).
- **Broken in comparison plots:** Same linestyle bug shows up in `ComparisonReport` of CV reports (`hue=estimator`, `style=data_source`, `units=split`).
- **Wrong legend colors:** Estimator colors in the legend can disagree with the actual curves (e.g. an estimator gets another estimator’s color).
- **Wrong legend order:** Legend entries may not follow the order estimators were passed to `ComparisonReport`.

**Comment by direkkakkar319-ops** [comment]
> maybe start with roc curve and precision recall since the fix should be similar. If prediction error is simple, you can add it too. FYI, here are the issues I found with data_source="both" for cv reports:
> 
> * **Inverted dash styles:** With `data_source="both"`, the legend shows train=dashed / test=solid, but the plotted curves often have it the other way around (especially on `CrossValidationReport`).
> * **Broken in comparison plots:** Same linestyle bug shows up in `ComparisonReport` of CV reports (`hue=estimator`, `style=data_source`, `units=split`).
> * **Wrong legend colors:** Estimator colors in the legend can disagree with the actual curves (e.g. an estimator gets another estimator’s color).
> * **Wrong legend order:** Legend entries may not follow the order estimators were passed to `ComparisonReport`.

Hi @GaetandeCast i am not able to reproduce the issue you mentioned can you help me out with that ,  is it based on a specific condition that i am not able to use ?

**Comment by GaetandeCast** [comment]
With this reproducer: 

```python
from sklearn.datasets import load_breast_cancer
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import HistGradientBoostingClassifier
from sklearn.tree import DecisionTreeClassifier
from skore import compare, evaluate

X, y = load_breast_cancer(return_X_y=True)
report = compare([
    evaluate(HistGradientBoostingClassifier(), X, y, splitter=2),
    evaluate(DecisionTreeClassifier(max_depth=1), X, y, splitter=2),
])

display = report.metrics.roc(data_source="both")
display.plot(subplot_by=None, label=0)
```
you can see that the color are inverted in the legend: HGBC is shown as linked to blue lines in the legend when it's obviously the orange lines in the plot and the style of dashes is inverted: the train lines are full on the plot but dashed in the legend.
Btw for the dashes we want the behavior of the legend: train as dashed and test as full.

**Comment by direkkakkar319-ops** [comment]
Thank you @GaetandeCast now will be able to to do it

**Comment by GaetandeCast** [comment]
Cool, don't hesitate to ping me when it's ready for review.

**Comment by GaetandeCast** [comment]
We still have `PredictionErrorDisplay` to fix.

**Comment by direkkakkar319-ops** [comment]
Hi @GaetandeCast i would love to move to the next part of the issue and work on `PredictionErrorDisplay `: )

**Comment by GaetandeCast** [comment]
Sure, go for it @direkkakkar319-ops !
